### PR TITLE
[Page] Expose components from Template in Page Model

### LINKED ISSUE #134: [Page] Expose components from Template in Page Model
Components of the template are currently also not exposed when requesting the page with .model.json selector.

### PR DESCRIPTION
**Pull request by vladbailescu**
* get the page model for the structured resources returned by the template manager
* unit tests

(fixes #134)

### CODE REVIEW

**Comment by pfauchere** [comment]
@vladbailescu could you please review the current PR?

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PageImpl.java [hunk 1]
```diff
@@ -241,7 +244,32 @@ public String getExportedType() {
...
+                TemplateManager templateManager = resolver.adaptTo(TemplateManager.class);
+
+                // Resources backed by the template structure
+                if (templateManager != null) {
+                    children = templateManager.getStructureResources(componentContext);
+                }
+            }
+        }
+
+        if (children == null || !children.iterator().hasNext()) {
```
Comment (raducotescu) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PageImpl.java]: @pfauchere, what's the intention here? It seems like if the template provides structure resources then the page will not expose any more its own components.

Comment (pfauchere) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PageImpl.java]: They will be returned by the `TemplateManager#getStructureResources` function extracting the content of the page using the `componentContext`. And extend that content with the one from the structure of the template.

If no structured resources are found, we fallback to the default

File: bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/PageImplTest.java [hunk 2]
```diff
@@ -194,53 +213,135 @@ public void testDefaultDesign() {
...
+        testGetChildModel(false);
+    }
+
+    @Test
+    public void testGetChildModelEditableTemplate() {
+        testGetChildModel(true);
+    }
+
+    @Test
+    public void testGetChildModelStructured() {
```
[the remark this answers is not in the document]
Comment (pfauchere) [reply bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/PageImplTest.java]: Leftover after the refactoring. Thx for the catch

**Pull request merged**